### PR TITLE
fix knowhere-hnsw filter strategy

### PR DESCRIPTION
**Pull request by alwayslove2013**
- update alpha from filter_rate / 2 to filter_rate * 0.7
- set the initial value of acc_alpha from 0.0 to 1.0
- switch to brute force if results are less than k.

issue: https://github.com/milvus-io/milvus/issues/34548

results: recall++, qps--
image

### CODE REVIEW

File: thirdparty/hnswlib/hnswlib/hnswalg.h [hunk 1]
```diff
@@ -1441,6 +1441,10 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
                 searchBaseLayerST<false, true>(currObj, query_data, std::max(ef, k), visited, bitset, feder_result);
         }
         std::vector<std::pair<dist_t, labeltype>> result;
+        // switch to brute-force when insufficient k
```
Comment (PwzXxm) [inline thirdparty/hnswlib/hnswlib/hnswalg.h:1444]: move L1444-1447 before L1443 plz

Comment (alwayslove2013) [reply thirdparty/hnswlib/hnswlib/hnswalg.h:1443]: got it~

**Pull request merged**